client/src/pages/Signin.js

```diff
@@ -0,0 +1,179 @@
+/** @format */
+
+import React, { useState, useEffect } from "react";
+import { Box, Container, Grid, Typography } from "@material-ui/core";
+import { TextField, Button } from "@material-ui/core";
+import { Link, Redirect } from "react-router-dom";
+import { useStyles } from "../themes/theme";
+
+const Signin = () => {
+	const classes = useStyles();
+
+	const [state, setState] = useState({
+		emailSignin: "",
+		pwdSignin: "",
+		emailError: false,
+		emailValidationError: false,
+		pwdError: false,
+		redirect: false,
+	});
+
+	useEffect(() => {
+		setState(state);
+	}, [state]);
+
+	const handleInputChange = (event) => {
+		setState({
+			...state,
+			[event.target.name]: event.target.value,
+			emailError: false,
+			emailValidationError: false,
+			pwdError: false,
+		});
+	};
+
+	const checkEmail = () => {
+		const email_patt = new RegExp(`[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,}$`);
+		const email_test = email_patt.test(state.emailSignin);
+		return email_test ? true : false;
+	};
+
+	const handleSubmit = (event) => {
+		event.preventDefault();
+		const { emailSignin, pwdSignin } = state;
+		const userInfo = {
+			email: emailSignin.toLowerCase(),
+			password: pwdSignin.toLowerCase(),
+		};
+		if (checkEmail() && pwdSignin) {
+			fetch("/api/users/login", {
+				method: "POST",
+				headers: {
+					"Content-Type": "application/json",
+				},
+				body: JSON.stringify(userInfo),
+			})
+				.then((res) => res.json())
+				.then((res) => {
+					if (res.status === 200) {
+						//redirect to dashboard
+						setState({ ...state, redirect: true });
+					} else if (res.status === []) {
+						setState({ ...state, emailError: true });
+					} else {
+						setState({ ...state, pwdError: true });
+					}
+				})
+				.catch();
+		} else if (!checkEmail()) {
+			setState({ ...state, emailValidationError: true });
+		} else if (!pwdSignin) {
+			setState({ ...state, pwdError: true });
+		}
+	};
+
+	return state.redirect ? (
+		<Redirect to="/dashboard" />
+	) : (
+		<Container id="modal-content" maxWidth="xs">
+			<Box textAlign="right" className="modal-header">
+				<Button size="small" className={classes.close}>
+					&times;
+				</Button>
```

Comment:
This could be made a little bit cleaner by declaring the whole `<Container id="modal-content" maxWidth...` component into a separate constant, so that the ternary operator is easy to read in one line or two. For example:

```javascript
const signInContainer = ( <Container id="modal-content" maxWidth="xs"> ... </Container>)

return state.redirect ? <Redirect to="/dashboard" /> : signInContainer;